tracer.go

```diff
@@ -73,6 +73,12 @@ type Options struct {
 	// When set, it attempts to exacerbate issues emanating from use of Spans
 	// after calling Finish by running additional assertions.
 	DebugAssertUseAfterFinish bool
+	// DisableSpanPool disables reuse of the underlying span structs.
+	// Buyer Beware, any operations called on a span after Finish is undefined
+	// behavior. The main usecase of disabling the pool is to allow the creation
```

Comment:
@jmacd said:

> I'd feel better if ... Span use-after-finish were permitted except when EnableSpanPool is set

I would like the Span impl to not-crash in that scenario, but I do not want to document any behavior about post-`Finish` semantics beyond that minimal safety guarantee. (I think @jmacd would agree with the above, but it is ambiguous in the quoted snippet just above)